pinot-core/src/main/java/org/apache/pinot/core/operator/CombineGroupByOperator.java

```diff
@@ -224,6 +241,8 @@ public void runJob() {
           future.cancel(true);
         }
       }
+      // Deregister the main thread and wait for all threads done
```

Comment:
In line 179, don't you need to issue a cacel on the futures, and then jump down to awaitAndDeregister()?

Reply:
OK, I see that the phaser. awaitAdvance() should cover this case. Thanks